src/library_webgl.js

```diff
@@ -2059,28 +2007,118 @@ var LibraryGL = {
     emscriptenWebGLGetUniform(program, location, params, {{{ cDefine('EM_FUNC_SIG_PARAM_I') }}});
   },
 
+  // Returns the WebGLUniformLocation object corresponding to the location index integer on
+  // the currently active shader in this GL context.
+  $webglGetUniformLocation: function(location) {
+    var p = GLctx.currentProgram;
+    var webglLoc = p.uniformLocsById[location];
+    // p.uniformLocsById[location] stores either an integer, or a WebGLUniformLocation.
+
+    // If an integer, we have not yet bound the location, so do it now. The integer value specifies the array index
+    // we should bind to.
+    if (webglLoc >= 0) {
+      p.uniformLocsById[location] = webglLoc = GLctx.getUniformLocation(p, p.uniformArrayNamesById[location] + (webglLoc > 0 ? '[' + webglLoc + ']' : ''));
+    }
+    // Else an already cached WebGLUniformLocation, return it.
+    return webglLoc;
+  },
+
   glGetUniformLocation__sig: 'iii',
   glGetUniformLocation__deps: ['$jstoi_q'],
   glGetUniformLocation: function(program, name) {
+    // Returns the index of '[' character in an uniform that represents an array of uniforms (e.g. colors[10])
+    // Closure does counterproductive inlining: https://github.com/google/closure-compiler/issues/3203, so prevent
+    // inlining manually.
+    /** @noinline */
+    function getLeftBracePos(name) {
+      return name.slice(-1) == ']' && name.lastIndexOf('[');
+    }
+
 #if GL_ASSERTIONS
     GL.validateGLObjectID(GL.programs, program, 'glGetUniformLocation', 'program');
 #endif
     name = UTF8ToString(name);
 
+#if GL_ASSERTIONS
+    assert(name.trim() == name, 'Uniform names passed to glGetUniformLocation() should not contain slop trimmable whitespace!');
```

Comment:
nit: perhaps go further with this assertion and check it matches some pattern? I think it must match something like `/^\w+(\[\d*\])?$/`. The name and size could even be extracted with this but maybe it takes more bytes.

Reply:
The intent here was not to perfectly validate what is a proper uniform name (we'll defer that to the WebGL backend), but to verify against the fact that we programmatically parse the `name` string in a way that would let whitespace slip through, i.e. this function does now potentially work for `colors[10] ` since we split at the first found `[`.

Reply:
I get that, but it seems like there are other weird things that could happen, like `colors [10]` or `colors[ 10 ]` which AFAICT would also not be valid. I wouldn't expect there's much cost to having a stricter assertion here.

Reply:
I see.. updated the check to verify that there are no whitespace.